source/adapters/level_zero/command_buffer.cpp

```diff
@@ -251,27 +303,33 @@ static ur_result_t enqueueCommandBufferMemCopyHelper(
     void *Dst, const void *Src, size_t Size, uint32_t NumSyncPointsInWaitList,
     const ur_exp_command_buffer_sync_point_t *SyncPointWaitList,
     ur_exp_command_buffer_sync_point_t *SyncPoint) {
-  std::vector<ze_event_handle_t> ZeEventList;
-  UR_CALL(getEventsFromSyncPoints(CommandBuffer, NumSyncPointsInWaitList,
-                                  SyncPointWaitList, ZeEventList));
-
-  ur_event_handle_t LaunchEvent;
-  UR_CALL(
-      EventCreate(CommandBuffer->Context, nullptr, false, false, &LaunchEvent));
-  LaunchEvent->CommandType = CommandType;
+  if (CommandBuffer->IsInOrderCmdList) {
+    ZE2UR_CALL(
+        zeCommandListAppendMemoryCopy,
+        (CommandBuffer->ZeCommandList, Dst, Src, Size, nullptr, 0, nullptr));
 
```

Comment:
wondering if it's an issue in these code paths than we never set the return `SyncPoint` parameter to the entry-point, we could maybe set it but not register it with the command-buffer

Reply:
Hmm, returning a sync-point value that is silently ignored seems a little unintuitive to me. 

Perhaps the issue is that we don't have a way to report that it is unused/not set? Since sync points are a typedef for `uint32_t` and `0` is a valid value for a sync point. If it was signed we could return `-1` to signify the sync point is unused, or swap the typedef for a simple struct/class that could indicate whether it is a valid sync point. This might be overkill though for this issue.

If we are ignoring sync points passed to these entry points it might also be better to make that explicit and return an error if the user tries to pass sync points when appending to an in-order command-buffer? That would make it more explicit but also increase the burden on the user side whether to store/pass the sync points or not.

Reply:
> If we are ignoring sync points passed to these entry points it might also be better to make that explicit and return an error if the user tries to pass sync points when appending to an in-order command-buffer? That would make it more explicit but also increase the burden on the user side whether to store/pass the sync points or not.

Agree with adding some spec wording as a solution, would make the obligations on caller/impl explicit. Would probably also be okay with the spec saying an in-order command-buffer will ignore any sync point parameters rather than making it a hard error if that makes some usage code simpler.